<?xml version="1.0" encoding="UTF-8"?><rss version="2.0"
	xmlns:content="http://purl.org/rss/1.0/modules/content/"
	xmlns:dc="http://purl.org/dc/elements/1.1/"
	xmlns:atom="http://www.w3.org/2005/Atom"
	xmlns:sy="http://purl.org/rss/1.0/modules/syndication/"
	
	>
<channel>
	<title>Comments on: Generic error handler for ExtJS</title>
	<atom:link href="http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/feed/" rel="self" type="application/rss+xml" />
	<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/</link>
	<description>Enterprise Software Development with JavaScript and Java</description>
	<lastBuildDate>Sat, 14 Nov 2015 03:24:43 +0000</lastBuildDate>
	<sy:updatePeriod>hourly</sy:updatePeriod>
	<sy:updateFrequency>1</sy:updateFrequency>
	<generator>http://wordpress.org/?v=4.3.1</generator>
	<item>
		<title>By: Ankit Garg</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-8154</link>
		<dc:creator><![CDATA[Ankit Garg]]></dc:creator>
		<pubDate>Wed, 12 Oct 2011 03:25:02 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-8154</guid>
		<description><![CDATA[Hi, thanks for this code, it helped me a lot. I have one question though, the error handler is fired if an ajax request is in progress and user navigates away from the page. The isAbort flag of the response object is also false in that case. Is there a way to determine that the reason for the failure is user navigating away from the page and not show the alert message in that case?]]></description>
		<content:encoded><![CDATA[<p>Hi, thanks for this code, it helped me a lot. I have one question though, the error handler is fired if an ajax request is in progress and user navigates away from the page. The isAbort flag of the response object is also false in that case. Is there a way to determine that the reason for the failure is user navigating away from the page and not show the alert message in that case?</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: batz</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-7559</link>
		<dc:creator><![CDATA[batz]]></dc:creator>
		<pubDate>Tue, 22 Feb 2011 15:08:48 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-7559</guid>
		<description><![CDATA[Thx for the post, very useful]]></description>
		<content:encoded><![CDATA[<p>Thx for the post, very useful</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Johan van de Merwe</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-6615</link>
		<dc:creator><![CDATA[Johan van de Merwe]]></dc:creator>
		<pubDate>Sat, 30 Oct 2010 11:49:09 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-6615</guid>
		<description><![CDATA[But I see that my response can be looking silly because of Ext version difference on the first article, I relate to version 3.3.

Johan]]></description>
		<content:encoded><![CDATA[<p>But I see that my response can be looking silly because of Ext version difference on the first article, I relate to version 3.3.</p>
<p>Johan</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Johan van de Merwe</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-6614</link>
		<dc:creator><![CDATA[Johan van de Merwe]]></dc:creator>
		<pubDate>Sat, 30 Oct 2010 11:47:04 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-6614</guid>
		<description><![CDATA[Hello Marcus,

Why not handling generic server errors with the exception event in Ext.data.DataProxy? We have run into the problem that our session expired after 2 hours (CodeIgniter) and that the user clicked in her application and empty grids appeared. This is handled fine by a 

Ext.data.DataProxy.on (&#039;exception&#039;, base.eventManager.onServerError);

where base.eventManager is our eventmanager for our base functionalities. 

Kind regards,

Johan]]></description>
		<content:encoded><![CDATA[<p>Hello Marcus,</p>
<p>Why not handling generic server errors with the exception event in Ext.data.DataProxy? We have run into the problem that our session expired after 2 hours (CodeIgniter) and that the user clicked in her application and empty grids appeared. This is handled fine by a </p>
<p>Ext.data.DataProxy.on (&#8216;exception&#8217;, base.eventManager.onServerError);</p>
<p>where base.eventManager is our eventmanager for our base functionalities. </p>
<p>Kind regards,</p>
<p>Johan</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Andres Serron</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-4297</link>
		<dc:creator><![CDATA[Andres Serron]]></dc:creator>
		<pubDate>Thu, 06 May 2010 18:57:23 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-4297</guid>
		<description><![CDATA[Nice code.

Ajax broadcasts 3 events that could also perform logic before any callback.

Options is available there so you could tap into the scope to work there.

But your code enlight me to think in some application where we could block further actions, like an aspect over the response.

Like the chain of events in the editorgrids when validation occurs.

cheers.]]></description>
		<content:encoded><![CDATA[<p>Nice code.</p>
<p>Ajax broadcasts 3 events that could also perform logic before any callback.</p>
<p>Options is available there so you could tap into the scope to work there.</p>
<p>But your code enlight me to think in some application where we could block further actions, like an aspect over the response.</p>
<p>Like the chain of events in the editorgrids when validation occurs.</p>
<p>cheers.</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Sam</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-622</link>
		<dc:creator><![CDATA[Sam]]></dc:creator>
		<pubDate>Wed, 01 Jul 2009 01:39:25 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-622</guid>
		<description><![CDATA[Hello Marcus:
   It&#039;s a very useful method for development. but does it work in ExtJS3?
   when I checked the code of Connection class, I found it will fire a &quot;REQUESTEXCEPTION&quot; event when request failed.  so can we use event mode to handle the error?]]></description>
		<content:encoded><![CDATA[<p>Hello Marcus:<br />
   It&#8217;s a very useful method for development. but does it work in ExtJS3?<br />
   when I checked the code of Connection class, I found it will fire a &#8220;REQUESTEXCEPTION&#8221; event when request failed.  so can we use event mode to handle the error?</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Marcus Schiesser</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-290</link>
		<dc:creator><![CDATA[Marcus Schiesser]]></dc:creator>
		<pubDate>Wed, 04 Mar 2009 21:09:33 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-290</guid>
		<description><![CDATA[Thanks Animal, that&#039;s even better!]]></description>
		<content:encoded><![CDATA[<p>Thanks Animal, that&#8217;s even better!</p>
]]></content:encoded>
	</item>
	<item>
		<title>By: Animal</title>
		<link>http://marcusschiesser.de/2009/02/09/generic-error-handler-for-extjs/comment-page-1/#comment-285</link>
		<dc:creator><![CDATA[Animal]]></dc:creator>
		<pubDate>Mon, 02 Mar 2009 16:33:25 +0000</pubDate>
		<guid isPermaLink="false">http://www.marcusschiesser.de/?p=148#comment-285</guid>
		<description><![CDATA[Check the Function methods Ext gives you. You have just effectively created an interceptor. The code I would have written would be

Ext.data.Connection.prototype.handleFailure = Ext.data.Connection.prototype.handleFailure.createInterceptor( function(response, e) {
	var errorText = Ext.DomQuery.selectValue(&quot;Reason/Text&quot;, response.responseXML, &quot;Unknown Error&quot;);
	Ext.Msg.alert(&#039;Error&#039;, errorText);
});]]></description>
		<content:encoded><![CDATA[<p>Check the Function methods Ext gives you. You have just effectively created an interceptor. The code I would have written would be</p>
<p>Ext.data.Connection.prototype.handleFailure = Ext.data.Connection.prototype.handleFailure.createInterceptor( function(response, e) {<br />
	var errorText = Ext.DomQuery.selectValue(&#8220;Reason/Text&#8221;, response.responseXML, &#8220;Unknown Error&#8221;);<br />
	Ext.Msg.alert(&#8216;Error&#8217;, errorText);<br />
});</p>
]]></content:encoded>
	</item>
</channel>
</rss>
